### ISSUE TITLE
Templates: Extra variables in templates and how does metatype fit in

### ISSUE THREAD

**Issue by zhsc** [bndtools/bndtools#1711]
I've noticed when using templates, at least with string template not sure about mustache, that unofficial variables are prompted for in the wizard. 

It seems there's some metatype stuff around making it a little more friendly than just the variable name that's possible but it's pretty obscure from just briefly looking at the code. 

Maybe the metatype could be explained on the readme where the rest of templates are explained. 

Also I believe string template supports extra attributes after the template include.  I.e. $ something;  extra="xyz"$ 

Not sure if that could be parseable with the interpreter that's looking for template variables to provide some extra data like labels or required indicator or default values. 

Just some thoughts. Which maybe should have gone on mailing lists are instead.  :-)

Let me know your thoughts. 

Thanks!

**Comment by njbartlett** [comment]
This sounds like a request to document the template and metatype system for the 3.4 release, which I agree with.

FWIW I would tend to encourage using Mustache over StringTemplate because it is much simpler and easier to understand.

**Comment by zhsc** [comment]
Yeah,  basically just some additional notes about the template system and how to do variables "right" in the vision of the template system.  

Thought I'd throw in the comment about the ST include directives if it'd help with an approach. 

It's really cool to be able to create a template in a repo and just have it show up in the new project wizard.  My boss has had lots of template requests recently since we've started using it.

Thanks
